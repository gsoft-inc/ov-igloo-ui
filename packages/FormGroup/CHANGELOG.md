# @igloo-ui/form-group

## 1.0.8

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.
- Updated dependencies [cfb5619]
  - @igloo-ui/helper-text@0.2.17

## 1.0.7

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.
- Updated dependencies [7d36815]
  - @igloo-ui/helper-text@0.2.16

## 1.0.6

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency
- Updated dependencies [a2df55d]
  - @igloo-ui/helper-text@0.2.15

## 1.0.5

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.
- Updated dependencies [9739155]
  - @igloo-ui/helper-text@0.2.14

## 1.0.4

### Patch Changes

- 24146bc: Gave the FormGroup component the new brand.

## 1.0.3

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.
- Updated dependencies [c3efaf5]
  - @igloo-ui/helper-text@0.2.10

## 1.0.2

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.
- Updated dependencies [a41e1d4]
  - @igloo-ui/helper-text@0.2.9

## 1.0.1

### Patch Changes

- 621de79: fix the error of the documentation in relation to the use of the `FormGroup`

## 1.0.0

### Major Changes

- 9b9ec61: Changed some props of form group to allow different types of messages to be added and not just errors

## 0.1.3

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.2

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2

## 0.1.1

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.1.0

### Minor Changes

- b8729ca: Feat: Initial release of the form group component
