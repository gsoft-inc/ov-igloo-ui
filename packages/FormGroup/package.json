{
  "name": "@igloo-ui/form-group",
  "version": "0.1.0",
  "main": "dist/FormGroup.js",
  "module": "dist/FormGroup.js",
  "types": "dist/FormGroup.d.ts",
  "styles": "dist/form-group.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^1.0.0",
    "@igloo-ui/icons": "^1.3.0",
    "classnames": "^2.3.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ],
  "devDependencies": {
    "@igloo-ui/input": "^1.0.6",
    "@igloo-ui/select": "^0.1.0",
    "@igloo-ui/textarea": "^0.1.1"
  }
}
