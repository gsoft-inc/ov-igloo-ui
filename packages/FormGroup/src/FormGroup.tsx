import * as React from 'react';

import cx from 'classnames';
import Alert from '@igloo-ui/icons/dist/Alert';

import './form-group.scss';

export interface FormGroupProps extends React.ComponentProps<'div'> {
  /** The form element that needs an error and/or a label. */
  children: React.ReactNode;
  /** Add a className for the form group div */
  className?: string;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** Add an error message below the form element */
  errorMsg?: string;
  /** Add the htmlFor attribute to the label of the form element */
  htmlFor?: string;
  /** Add label text above the form element */
  label?: string;
  /** Decides when to show the error */
  showError?: boolean;
}

const FormGroup: React.FunctionComponent<FormGroupProps> = (
  props: FormGroupProps
) => {
  const { children, className, dataTest, errorMsg, htmlFor, label, showError } =
    props;

  const classes = cx('ids-form-group', className);

  return (
    <div className={classes} data-test={dataTest}>
      {label && (
        <label htmlFor={htmlFor} className="ids-form-group__label">
          {label}
        </label>
      )}
      {children}
      {errorMsg && showError && (
        <div className="ids-form-group__error">
          <Alert size="small" />
          {errorMsg}
        </div>
      )}
    </div>
  );
};

export default FormGroup;
