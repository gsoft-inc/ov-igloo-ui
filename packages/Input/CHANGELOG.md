# @igloo-ui/input

## 1.1.0

### Minor Changes

- 4a388ae: Add Postfix icon position in Input component

## 1.0.7

### Patch Changes

- 8bc7f2c: remove focus border on disabled state

## 1.0.6

### Patch Changes

- 38a4994: Update dependency on @igloo-ui/icons and other @igloo libs

## 1.0.5

### Patch Changes

- 0a86f13: Update content of README

## 1.0.4

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 1.0.3

### Patch Changes

- 4a256e3: IE11 compatibility

## 1.0.2

### Patch Changes

- 519f009: Uncontrolled input bugfix

## 1.0.1

### Patch Changes

- 2da8010: Fix render of elements with the css var --grey-{...}

## 1.0.0

### Major Changes

- ef50a1b: BRAKING CHANGE

  - Replace `readOnly` props by `disabled`

  New Features

  - Add options for passing a Icon on the input

  Update

  - Upgrade tokens version

## 0.1.0

### Minor Changes

- 8e6a4ec: ## Input component

  The Input component specifies an input field where the user can enter data.

  ### Component API

  | name      | description                                          |
  | --------- | ---------------------------------------------------- |
  | type      | Specifies the type to display                        |
  | isCompact | True for a compact appearance.                       |
  | value     | Specifies the value inside the input.                |
  | autoFocus | True if you need the input to be focus on page load. |
  | readOnly  | True if you need the input to be readonly.           |
  | onChange  | Function called when a key is pressed.               |
  | dataTest  | Add a data-test tag for automated tests              |
