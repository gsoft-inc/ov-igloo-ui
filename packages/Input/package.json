{
    "name": "@igloo-ui/input",
    "version": "2.3.2",
    "main": "dist/Input.js",
    "module": "dist/Input.js",
    "types": "dist/Input.d.ts",
    "styles": "dist/input.css",
    "publishConfig": {
        "access": "public"
    },
    "repository": "git+https://github.com/workleap/ov-igloo-ui.git",
    "author": "Officevibe Inc.",
    "contributors": [
        "Sylvester sylvester.vuong@gsoft.com"
    ],
    "license": "Apache-2.0",
    "peerDependencies": {
        "react": ">= 16.8.6"
    },
    "files": [
        "dist"
    ],
    "scripts": {
        "build": "rollup -c rollup.config.js"
    },
    "dependencies": {
        "@igloo-ui/tokens": "^2.1.0",
        "@hopper-ui/tokens": "^4.3.2",
        "classnames": "^2.3.2",
        "graphemer": "^1.4.0"
    },
    "devDependencies": {
        "@igloo-ui/icons": "^1.12.2"
    },
    "browserslist": [
        "> 1%",
        "last 2 versions",
        "IE >= 11"
    ]
}
