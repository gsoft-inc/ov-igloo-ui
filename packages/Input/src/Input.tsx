import React, { useEffect } from 'react';
import cx from 'classnames';

import './input.css';

const KeyCodes = {
  Comma: 44,
  Minus: 45,
  Nine: 57,
  Period: 46,
  Plus: 43,
  Zero: 48,
};

export type InputSize = 'small';

export type InputType = 'email' | 'text' | 'password' | 'number';

export interface InputProps extends React.ComponentPropsWithRef<'input'> {
  // Add class names to the surrounding DOM container.
  className?: string;
  // Form.ValidatedField state. True if has error.
  error?: boolean;
  // Specifies the type to display
  type?: InputType;
  // Input size.
  inputSize?: InputSize;
  // Specifies the value inside the input.
  value?: string | number;
  // True if you need the input to be focus on page load.
  autoFocus?: boolean;
  // True if you need the input to be readonly.
  readOnly?: boolean;
  // Function called when the value changes.
  onChange?: (e: any) => void;
  // Add a data-test tag for automated tests
  dataTest?: string;
}

const Input: React.FunctionComponent<InputProps> = React.forwardRef(
  (props: InputProps, ref: React.Ref<any>) => {
    const {
      className,
      error,
      type = 'text',
      inputSize,
      value,
      autoFocus,
      readOnly,
      onChange,
      dataTest,
      ...rest
    } = props;

    const classes = cx(
      'ids-input',
      className,
      inputSize && `ids-input--${inputSize}`,
      error,
      {
        'ids-input--readonly': readOnly,
      }
    );

    useEffect(() => {
      const refObject = ref as React.RefObject<HTMLInputElement>;
      if (autoFocus && refObject && refObject.current) {
        refObject.current.focus();
      }
    }, [ref, autoFocus]);

    const handleOnChange = (e: any) => {
      if (onChange) {
        onChange(e);

        return;
      }

      // Firefox doesn't support type="number", so here's a workaround
      // Input type="number" on Chrome supports "+", "-", "," et "."
      if (type === 'number') {
        const { charCode } = e.charCode;

        if (
          charCode !== KeyCodes.Comma &&
          charCode !== KeyCodes.Period &&
          charCode !== KeyCodes.Minus &&
          charCode !== KeyCodes.Plus &&
          (charCode < KeyCodes.Zero || charCode > KeyCodes.Nine)
        ) {
          e.preventDefault();
        } else if (
          (charCode === KeyCodes.Comma || charCode === KeyCodes.Period) &&
          value &&
          typeof value === 'string' &&
          (value.indexOf(',') !== -1 || value.indexOf('.') !== -1)
        ) {
          e.preventDefault();
        } else if (
          (charCode === KeyCodes.Minus || charCode === KeyCodes.Plus) &&
          value &&
          typeof value === 'string' &&
          (value.indexOf('-') !== -1 || value.indexOf('+') !== -1)
        ) {
          e.preventDefault();
        }
      }
    };

    return (
      <input
        ref={ref}
        className={classes}
        type={type}
        value={value}
        readOnly={readOnly}
        onChange={handleOnChange}
        data-test={dataTest}
        {...rest}
      />
    );
  }
);

export default Input;
