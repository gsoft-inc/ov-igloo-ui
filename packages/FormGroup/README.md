# FormGroup

The FormGroup is a component that displays a label and an error message for a form input.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/form-group` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/form-group
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/form-group
```

## Usage

Then to use the component in your code just import it!

```jsx
import FormGroup from '@igloo-ui/form-group';
import Input from '@igloo-ui/input';

<FormGroup label="Name" errorMsg="Name is required" showError={true}>
  <Input type="text" placeholder="John Doe" error={true} />
</FormGroup>;
```
