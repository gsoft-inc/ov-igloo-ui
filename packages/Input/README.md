# Input

Input enable the user to interact with and used for freeform data entry.

<ReferenceLinks />

## Installation

To install `@igloo-ui/input` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/input
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/input
```

## Usage

Then to use the component in your code just import it!

```jsx
import Input from '@igloo-ui/input';

<Input type="text" placeholder="input type text" />;
```
