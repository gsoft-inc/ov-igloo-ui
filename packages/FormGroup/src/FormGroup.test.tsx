/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import FormGroup, { FormGroupProps } from './FormGroup';
import Input from '@igloo-ui/input';

const setup = (props: FormGroupProps) => {
  return shallow(<FormGroup {...props}></FormGroup>);
};

describe('FormGroup', () => {
  const inputField = <Input type="text" placeholder="Enter your full name" />;
  const component = () => setup({ children: inputField });

  test('It should render without errors', () => {
    const wrapper = component().find('.ids-form-group');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component()).toMatchSnapshot();
  });

  test('It should render a label', () => {
    const formGroupComponent = () =>
      setup({ children: inputField, label: 'I am a label' });
    const wrapper = formGroupComponent().find('.ids-form-group__label');
    expect(wrapper.length).toBe(1);
  });

  test('It should render an error', () => {
    const formGroupComponent = () =>
      setup({
        children: inputField,
        errorMsg: 'I am an error',
        showError: true,
      });
    const wrapper = formGroupComponent().find('.ids-form-group__error');
    expect(wrapper.length).toBe(1);
  });
});
