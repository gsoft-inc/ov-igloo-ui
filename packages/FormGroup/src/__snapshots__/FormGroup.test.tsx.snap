// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup It should render a snapshot 1`] = `ShallowWrapper {}`;
